spring-integration-websocket/src/test/java/org/springframework/integration/websocket/client/StompIntegrationTests.java

```diff
@@ -350,7 +353,9 @@ public IntegrationWebSocketContainer clientWebSocketContainer() {
 
 		@Bean
 		public SubProtocolHandler stompSubProtocolHandler() {
-			return new StompSubProtocolHandler();
+			StompSubProtocolHandler stompSubProtocolHandler = new StompSubProtocolHandler();
+			stompSubProtocolHandler.setEncoder(new ClientStompEncoder());
+			return stompSubProtocolHandler;
 		}
```

Comment:
Shouldn't we use a different protocol handler on the client/server side? This should only be used on the client side, right?

Also, the test is checking the switch occurred on the server side, which seems wrong to me.

Surely, on the inbound side, we are expecting a `MESSAGE` frame.

By the way, these `sendMessageToController*` tests always fail for me in my IDE; it seems the event listener is not registered/started...

>15:18:58.747 TRACE [WebSocketClient-AsyncIO-1][org.springframework.context.support.GenericApplicationContext] Publishing event in org.springframework.context.support.GenericApplicationContext@2d2e5f00: SessionConnectedEvent[GenericMessage [payload=byte[0], headers={simpMessageType=OTHER, stompCommand=CONNECTED, nativeHeaders={heart-beat=[0,0]}, simpSessionAttributes={}, simpHeartbeat=[J@1bad8bb9, simpSessionId=53826b559db84ae3afe7e3a9309ba9ae}]]
>15:19:18.712 TRACE [main][org.springframework.integration.channel.QueueChannel] postReceive on channel 'webSocketEvents', message is null


Works in gradle...

>15:18:28.376 TRACE [WebSocketClient-AsyncIO-1][org.springframework.context.support.GenericApplicationContext] Publishing event in org.springframework.context.support.GenericApplicationContext@3aa3b61a: SessionConnectedEvent[GenericMessage [payload=byte[0], headers={simpMessageType=OTHER, stompCommand=CONNECTED, nativeHeaders={heart-beat=[0,0]}, simpSessionAttributes={}, simpHeartbeat=[J@21741628, simpSessionId=06af94fa8ce54e14afdc9bd185433e51}]]
15:18:28.377 DEBUG [WebSocketClient-AsyncIO-1][org.springframework.beans.factory.support.DefaultListableBeanFactory] Returning cached instance of singleton bean 'webSocketEventListener'
15:18:28.377 DEBUG [WebSocketClient-AsyncIO-1][org.springframework.integration.channel.QueueChannel] preSend on channel 'webSocketEvents', message: GenericMessage [payload=SessionConnectedEvent[GenericMessage [payload=byte[0], headers={simpMessageType=OTHER, stompCommand=CONNECTED, nativeHeaders={heart-beat=[0,0]}, simpSessionAttributes={}, simpHeartbeat=[J@21741628, simpSessionId=06af94fa8ce54e14afdc9bd185433e51}]], headers={id=f1ba3bbd-21d0-21fe-8933-d411ae4f26b6, timestamp=1509391108377}]
15:18:28.377 DEBUG [Test worker][org.springframework.integration.channel.QueueChannel] postReceive on channel 'webSocketEvents', message: GenericMessage [payload=SessionConnectedEvent[GenericMessage [payload=byte[0], headers={simpMessageType=OTHER, stompCommand=CONNECTED, nativeHeaders={heart-beat=[0,0]}, simpSessionAttributes={}, simpHeartbeat=[J@21741628, simpSessionId=06af94fa8ce54e14afdc9bd185433e51}]], headers={id=f1ba3bbd-21d0-21fe-8933-d411ae4f26b6, timestamp=1509391108377}]

...investigating...

Reply:
> Shouldn't we use a different protocol handler on the client/server side? 

We do. This `StompIntegrationTests` has `ContextConfiguration` - for client side. See there `clientWebSocketContainer`  bean for both channel adapters. And there is a separate `ServerConfig` picked up by the `TomcatWebSocketTestServer` bean definition.

>  the test is checking the switch occurred on the server side, which seems wrong to me.

The test indeed checks the state on the server side. See controller usage:
```
SimpleController controller = this.serverContext.getBean(SimpleController.class);
```
The goal is to be sure that client has sent the proper frame and server accepts it without any modifications.

> Surely, on the inbound side, we are expecting a MESSAGE frame.

Not sure what you mean since we test on server messages sent by the web socket client.

> By the way, these sendMessageToController* tests always fail for me in my IDE;

No, for me it's fine on the Windows...